metadata-ingestion/src/datahub/ingestion/source/s3/source.py

```diff
@@ -411,7 +411,9 @@ def get_fields(self, table_data: TableData, path_spec: PathSpec) -> List:
                     max_rows=self.source_config.max_rows
                 ).infer_schema(file)
             elif extension == ".json":
-                fields = json.JsonInferrer().infer_schema(file)
+                fields = json.JsonInferrer(
+                    is_newline_json=True    
```

Comment:
Should be 

```diff
- is_newline_json=True
+ is_newline_json=self.source_config.is_newline_json
```

Reply:
Please add the `newlinejson` as source's pip dependencies in the setup script as well.

Reply:
Thanks, @duyet. My bad :) PR was updated.